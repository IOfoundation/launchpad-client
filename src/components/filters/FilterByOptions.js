import React from 'react';
import {PropTypes} from 'prop-types';
import onClickOutside from 'react-onclickoutside';
import ArrowRight from '../shared/ArrowRight';

class FilterByOptions extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      dropdownOpen: false,
      subDropdownOpen: false,
      selectedfilterOption: null,
    };
  }
  _toggleDropdownOptions() {
    this.setState({dropdownOpen: !this.state.dropdownOpen});
  }
  _toggleSubOption(event, selectedfilterOption) {
    if (selectedfilterOption.children.length > 0) {
      this.setState({selectedfilterOption, subDropdownOpen: true});
    }
  }
  _onClick(selectedfilterOption) {
    this.props.handleOnChangeFilterOptions(selectedfilterOption.name, 'category');
  }
  handleClickOutside() {
    this.setState({
      dropdownOpen: false,
      subDropdownOpen: false,
      selectedfilterOption: null,
    });
  }
  filterBtnOption(filterOption) {
    return (
      <button
        className="dropdown-options"
        key={filterOption.id}
        onClick={e => this._onClick(filterOption)}
        onMouseOver={e => this._toggleSubOption(e, filterOption)}
      >
        <span>{filterOption.name}</span>
        {filterOption.children.length > 0 && (
          <ArrowRight className="dropdown-options-icon" size="20" />
        )}
      </button>
    );
  }
  _renderSubOptions(filterOption) {
    return (
      <div className="dropdown-sub-container">
        {filterOption.children.map(child => (
          <button
            className="dropdown-options"
            key={child.id}
            onClick={() => this.props.handleOnChangeFilterOptions(child.name, 'category')}
          >
            <span>{child.name}</span>
          </button>
        ))}
      </div>
    );
  }
  _renderOptions() {
    return (
      <div
        className={
          this.state.subDropdownOpen
            ? 'dropdown-container dropdown-container-expand' : 'dropdown-container'
        }
      >
        <div
          className={
            this.state.subDropdownOpen
              ? 'dropdown-btn-half' : 'dropdown-btn-full'
          }
        >
          {this.props.filterOptions.map(filterOption => (
            <div key={filterOption.id}>
              {this.filterBtnOption(filterOption)}
            </div>
          ))}
        </div>
        {this.state.selectedfilterOption &&
          this._renderSubOptions(this.state.selectedfilterOption)}
      </div>
    );
  }
  render() {
    return (
      <div className="col-md-3 col-xs-3 no-padding filter-btn-container text-xs-margin">
        <button
          className="filter-btn"
          onClick={() => this._toggleDropdownOptions()}
        >
          {this.props.filterName}
          <img
            className="filter-btn-icon"
            src="/static-data/images/Dropdown-arrow.svg"
          />
        </button>
        {this.state.dropdownOpen && this._renderOptions()}
      </div>
    );
  }
}

FilterByOptions.propTypes = {
  filterName: PropTypes.string.isRequired,
  filterOptions: PropTypes.array.isRequired,
  handleOnChangeFilterOptions: PropTypes.func,
};
export default onClickOutside(FilterByOptions);
