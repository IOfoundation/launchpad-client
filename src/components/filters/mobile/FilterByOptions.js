import React from 'react';
import {PropTypes} from 'prop-types';
import FilterOption from './FilterOption';
import ArrowRight from '../../shared/ArrowRight';

const FilterByOptions = ({filterOptions}) => {
  return (
    <div className="dropdown-input-container">
      <span className="filter-icon">
        <img src="../static-data/images/filter.png" />
        {'Filters'}
      </span>
      <input className="dropdown-input" id="filters-dropdown" type="checkbox" />
      <label className="dropdown-label" htmlFor="filters-dropdown">
        {'No filters selected'}
      </label>
      <ArrowRight
        className="right-arrow"
        size={20}
        style={{color: '#fff', verticalAlign: 'middle'}}
        htmlFor="filters-dropdown"
      />
      <label htmlFor="filters-dropdown" className="close-icon">
        <img src="/static-data/images/close.png" />
      </label>
      <div className="dropdown-container-filters">
        <FilterOption
          filterName={'Business Services'}
          filterOptions={filterOptions.businessServices}
          handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
        />
        <FilterOption
          filterName={'Industry'}
          filterOptions={filterOptions.industries}
          handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
        />
        <FilterOption
          filterName={'Business Stage'}
          filterOptions={filterOptions.stages}
          handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
        />
        <FilterOption
          filterName={'Underserved Com...'}
          filterOptions={filterOptions.communities}
          handleOnChangeFilterOptions={this.props.handleOnChangeFilterOptions}
        />
      </div>
    </div>
  );
};
FilterByOptions.propTypes = {
  filterOptions: PropTypes.arrayOf(PropTypes.object).isRequired,
  handleOnChangeFilterOptions: PropTypes.func.isRequired,
};

export default FilterByOptions;
