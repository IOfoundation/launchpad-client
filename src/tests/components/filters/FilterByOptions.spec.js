// TODO: test _renderOptions & _renderSubOptions, test _toggleDropdownOptions

import React from 'react';
import {shallow} from 'enzyme';

import FilterByOptions from '../../../components/filters/FilterByOptions.js';

const handleOnChangeFilterOptions = jest.fn();

function filters() {
  return [
    {
      id: 2,
      name: 'Financial Management',
      children: [
        {
          id: 3,
          name: 'Bookkeeping/Accounting Software Training',
        },
        {
          id: 4,
          name: 'Budget, Financial Statement & Cash Flow Training',
        },
      ],
    },
    {
      id: 10,
      name: 'Capital',
      children: [
        {
          id: 11,
          name: 'Accelerator Programs',
        },
        {
          id: 12,
          name: 'Equity Investors: Angels',
        },
      ],
    },
  ];
}

// function items() {
//   return [];
// }

// describe('<FilterByOptions />', () => {
//   it('_renderOptions should return a dropdown menu with 2 options', () => {
//     const wrapper = shallow(
//       <FilterByOptions
//         className={'filterByOptions'}
//         filterName={'Business Services'}
//         filterOptions={filters()}
//         handleOnChangeFilterOptions={handleOnChangeFilterOptions}
//       />
//     );
//     const dropdownButton = wrapper.find('.filter-btn text-thin');
//     expect(wrapper.find('.dropdown-options').length).toEqual(1);
//   });
// });
