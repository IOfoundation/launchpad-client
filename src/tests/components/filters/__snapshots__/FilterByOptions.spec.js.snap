// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<FilterByOptions /> renders snapshot of FilterByOptions 1`] = `
<FilterByOptions
  disableOnClickOutside={[Function]}
  enableOnClickOutside={[Function]}
  eventTypes={
    Array [
      "mousedown",
      "touchstart",
    ]
  }
  filterName="Business Services"
  filterOptions={
    Array [
      Object {
        "children": Array [
          Object {
            "id": 3,
            "name": "Bookkeeping/Accounting Software Training",
          },
          Object {
            "id": 4,
            "name": "Budget, Financial Statement & Cash Flow Training",
          },
        ],
        "id": 2,
        "name": "Financial Management",
      },
      Object {
        "children": Array [
          Object {
            "id": 11,
            "name": "Accelerator Programs",
          },
          Object {
            "id": 12,
            "name": "Equity Investors: Angels",
          },
        ],
        "id": 10,
        "name": "Capital",
      },
    ]
  }
  handleOnChangeFilterOptions={[Function]}
  outsideClickIgnoreClass="ignore-react-onclickoutside"
  preventDefault={false}
  stopPropagation={false}
/>
`;
